tests-pl/dif_tests.pl

```diff
@@ -0,0 +1,235 @@
+/**/
```

Comment:
I don't see anything using this file. The files in this folder are used by the rust test in the sibling tests/ folder and not automatically picked up.

Reply:
Yes, these seem to be syntax conformity tests. I just didn't know where else to put it. The tests aren't needed for this, but I wanted to make them available somehow. I put it in another commit so I can just remove it if people think it's better. However I think it would be very nice to have this kind of library test included somehow in the test process of Scryer.

If you want to run it manually, the command is something like:

```
$ scryer-prolog -g main dif_tests.pl
```

You can copy this file and run it with the current master or another version of Scryer and it will report this:

```
Failed tests:
"dif#12"
"dif#21"
"dif#o3"
"dif#12 but with multiple variables in the residuals"
```

Which are exactly the 3 tests that Scryer fails [here](https://www.complang.tuwien.ac.at/ulrich/iso-prolog/dif) (plus one I made that is a generalization of [dif#12](https://www.complang.tuwien.ac.at/ulrich/iso-prolog/dif#12)).

Reply:
What I meant was that a test similar to the [iso_conformity_tests test](https://github.com/mthom/scryer-prolog/blob/master/tests/scryer/src_tests.rs#L73-L77) should be added for this so that it is checked via `cargo test` and in turn in CI,
so that regressions are caught.

Reply:
Oh, so _that's_ where the library tests are. Ok, I'm going to add it there. Thank you.